model-optimizer/extensions/front/tf/embedding_segments_sum.py

```diff
@@ -162,15 +192,43 @@ def replace_sub_graph(self, graph: Graph, match: dict):
         gather = match['gather']
         select = match['select']
         where0 = match['where0']
-
-        log.debug('Found ExperimentalSparseWeightedSum2 pattern after {} with name {}'.format(sparse_fill_empty_rows.op, sparse_fill_empty_rows.name))
-
-        sparse_weighted_sum = ExperimentalSparseWeightedSum(graph, {'name': sparse_fill_empty_rows.name + '/ExperimentalSparseWeightedSum_'}).create_node()
-        gather0_1.in_port(0).get_connection().set_destination(sparse_weighted_sum.in_port(0))
-        greaterequal0.in_port(0).get_connection().set_destination(sparse_weighted_sum.in_port(1))
-        identity_spw.in_port(0).get_connection().set_destination(sparse_weighted_sum.in_port(2))
-        gather.in_port(0).get_connection().set_destination(sparse_weighted_sum.in_port(3))
-        sparse_fill_empty_rows.in_port(3).get_connection().set_destination(sparse_weighted_sum.in_port(4))
+        output_node_name = select.soft_get('name', select.id)
+
+        log.debug('Found EmbeddingSegmentsSum2 pattern after {} with name {}'.format(sparse_fill_empty_rows.op,
+                                                                                     sparse_fill_empty_rows.name))
+
+        split_for_indices = create_op_with_const_inputs(graph, Split, {1: int64_array(1)},
+                                                        {'num_splits': 2,
+                                                         'name': output_node_name + '/SplitForIndices'})
+        squeeze_for_indices = create_op_with_const_inputs(graph, Squeeze, {1: int64_array([1])})
+        split_for_dense_shape = create_op_with_const_inputs(graph, Split, {1: int64_array(0)},
+                                                            {'num_splits': 2,
+                                                             'name': output_node_name + '/SplitForDenseShape'})
+        squeeze_to_scalar = create_op_with_const_inputs(graph, Squeeze, {1: int64_array([0])})
+        cast_default_value = Cast(graph, {'name': output_node_name + '/CastDefaultValue', 'dst_type': np.int32}).create_node()
+        cast_num_segments = Cast(graph, {'name': output_node_name + '/CastSegmentsNumber', 'dst_type': np.int32}).create_node()
+        embedding_segments_sum = EmbeddingSegmentsSum(graph, {'name': output_node_name}).create_node()
+        rename_nodes([(select, output_node_name + '/AbandonedName'), (embedding_segments_sum, output_node_name)])
+
+        # connect parameters table
+        gather.in_port(0).get_connection().set_destination(embedding_segments_sum.in_port(0))
+        # connect indices values
+        greaterequal0.in_port(0).get_connection().set_destination(embedding_segments_sum.in_port(1))
+        # split and connect segment ids
+        gather0_1.in_port(0).get_connection().set_destination(split_for_indices.in_port(0))
+        squeeze_for_indices.in_port(0).connect(split_for_indices.out_port(0))
+        embedding_segments_sum.in_port(2).connect(squeeze_for_indices.out_port(0))
```

Comment:
Need to cast segment_ids to I32 as well. And remove sub-graph "Const->SplitFakeOutput".

Reply:
Resolved these issues. Now Wide and Deep works with EmbeddingSegmentsSum and IE inference matches with reference results.